internal/bodyprocessors/json.go

```diff
@@ -77,7 +90,11 @@ func readItems(json gjson.Result, objKey []byte, res map[string]string) {
 		var val string
 		switch value.Type {
 		case gjson.JSON:
-			readItems(value, objKey, res)
+			// call recursively with one less item to avoid doing infinite recursion
+			iterationError = readItems(value, objKey, maxRecursion-1, res)
```

Comment:
shouldn't it be `if maxRecursion == -1 { return maxRecursion } else { return maxRecursion -1 }`

Reply:
The `readItems` func is called always with the `maxRecursion` value > 0. Then each iteration will decrement this value. I'm not following your question :/